server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -4024,7 +4025,11 @@ private SnapshotsInProgress createSnapshot(
                     }
                 }
                 if (missing.isEmpty() == false) {
-                    throw new SnapshotException(snapshot, "Indices don't have primary shards " + missing);
+                    throw new SnapshotException(snapshot, Strings.format("""
+                        the following indices have unassigned primary shards \
+                        and cannot be included in a snapshot with [partial] set to [false]: %s; \
```

Comment:
`partial` defaults to `false` so that users get this behaviour without explicitly "set" it to `false`. I wonder whether we should provide some extra hint here?